spring-integration-core/src/main/java/org/springframework/integration/config/xml/AbstractChannelAdapterParser.java

```diff
@@ -73,6 +73,13 @@ protected final AbstractBeanDefinition parseInternal(Element element, ParserCont
 		if (StringUtils.hasText(phase)) {
 			propertyValues.add("phase", new TypedStringValue(phase));
 		}
+		String role = element.getAttribute("role");
```

Comment:
I don't see the same for inter-flow endpoints - `AbstractConsumerEndpointParser`.
As well as in the XML test below.
Although you cover that with `@Role`.

Any comments ?